packages/blade/src/components/Box/Box.tsx

```diff
@@ -32,15 +32,13 @@ const validateBackgroundColor = (
 /**
  * This function is to filter out any unexpected props passed by the user
  */
-const useValidatedBoxProps = (props: BoxProps): KeysRequired<BoxProps> => {
-  React.useEffect(() => {
-    validateBackgroundColor(props.backgroundColor);
-  }, [props.backgroundColor]);
-
+const getOnlyBoxProps = (props: BoxProps): KeysRequired<BoxProps> => {
```

Comment:
Moved the validation out of this to keep it `get` function that can be re-used in storybook args generation as well.